### ISSUE TITLE
Dashboard doesn't update/gather data from saves while I play.

### ISSUE THREAD

**Issue by MyrddinE** [eliasdoehne/stellaris-dashboard#53]
The dashboard (dashboard.exe) runs a web server, but does not seem to process save files live while I'm playing so it seems... useless? When I visit it using the internal browser, it complains that it's not the -alpha variant. I can't find any newer download to fetch. And the other executable (parse_saves.exe) only runs once and doesn't keep processing new data as I play.

So while it works, technically, it is useless because I have one data point instead of a beautiful timeline. Am I doing something wrong? Is there a .config option I'm missing?

**Comment by eliasdoehne** [comment]
Hi MyrddinE, Sorry I only saw this issue now, I had my github notifications disabled accidentally :roll_eyes:. 

Just in case it's still relevant:

In the settings page (or by editing config.yaml) you can adjust the directory that is scanned for save files. But since it seems to have picked up one of them I am not sure if that is actually causing the problem. Is there any indication of an error in the dashboard window? On startup it should show what value it is using, something like
```
2021-08-01 12:36:44,863 - stellarisdashboard.config - INFO - Updated setting save_file_path                   None -> /home/elias/.local/share/Paradox Interactive/Stellaris/save_backup/save games
```

Can you check if that path actually contains the save files on your system? And if not, override the setting to the correct location.

**Comment by MyrddinE** [comment]
I never saw your reply until now either, so who am I to complain? :smile:

When I run the dashboard.exe executable I get the updating settings line, pointing to the correct location. The issue is that it never tries to scan saves. When I launch a game it starts up 4 more threads (I can see the interleaved copies launching in the logs), and it warns me about using a development web server in a production deployment :grin: but there are never any log entries as I'm playing showing that it's even trying to scan the saves.

I'm playing ironman mode.

**Comment by eliasdoehne** [comment]
That is odd, I have had other people report this as well but never managed to reproduced it myself. Maybe it is related to ironman, just for testing could you confirm if the problem exists also on a regular save without ironman? Especially since you mentioned that you get one data point it sounds like the ironman save is either not getting overwritten or the application does not notice it. I will also try to run an ironman game for testing, maybe something has changed there (it used to work well but I have not tested it in a long time)

**Comment by eliasdoehne** [comment]
For me it seems to work well, I get a save parsed every 3 months in-game, so it's not related to ironman. Could you share the actual save file path that you see in the configuration during the start?

**Comment by MyrddinE** [comment]
Yesterday I played for 10 or 11 hours (lots of pause time during the day, as I was also 'working', so I made it through only about 2 years in 8 hours). The dashboard opened up, had the interleaved thread-start logs, and then only TWO parses all day, at 6:58pm and 6:59pm when I stopped playing.

<img width="931" alt="Screenshot 2022-01-27 174226" src="https://user-images.githubusercontent.com/435613/151461620-06412849-eee2-4f1f-b22e-ae1b762b3290.png">

So, to be clear... both the dashboard.exe and my game were open all day, and I played for hours before this pair of parses at ~7pm, but it never parsed during gameplay at all.

<img width="692" alt="Screenshot 2022-01-27 174717" src="https://user-images.githubusercontent.com/435613/151462038-0f3eacba-a4a4-4702-8c45-4bb2ebe23821.png">

**Comment by MyrddinE** [comment]
Found the issue! Cloud saves are located in a completely different location.

C:\Steam\userdata\<redacted>\281990\remote\save games

Even if I have cloud saving turned off for a particular save, the interim/autosaves all go in the cloud folder. So the confusion was caused because I could see the saves in the expected location (under Documents/Paradox/etc) but nothing ever happened until I quit. The fact that there were saves in the Documents folder meant I didn't go looking for further folders or check to see if there were alternate places the saves could be.

It was further confusing because I always play Ironman (to remove the temptation to save scum). It wasn't until I tried a non-ironman game, saw that there were many autosaves available to load from, but none were showing in the Documents folder that I researched more about save locations and discovered the Steam cloud-save folder.

Sorry for the confusion. It may be helpful to document this behavior in the FAQ.
